sugondat-chain/runtimes/sugondat-kusama/src/lib.rs

```diff
@@ -267,14 +274,244 @@ impl pallet_balances::Config for Runtime {
+//pub type SlowAdjustingFeeUpdate<R> = TargetedFeeAdjustment<
+//    R,
+//    TargetBlockFullness,
+//    AdjustmentVariable,
+//    MinimumMultiplier,
+//    MaximumMultiplier,
+//>;
+
+/// Currently pallet_transaction_payment use the following formula:
+///
+/// ```
+/// inclusion_fee = base_fee + length_fee + [targeted_fee_adjustment * weight_fee];
+/// ```
+///
+/// Letting us able to update `targeted_fee_adjustment` at the end of each block
+/// thanks to `FeeMultiplierUpdate`, this associated type is called inside the `on_finalize`
+/// of the transaction_payment pallet with the aim of converting the before `targeted_fee_adjustment`
+/// to a new one based on the congestion of the network
+///
+/// What this struct does is this PLUS a side effect, the goal is to reach a different formula to
+/// calculate fees:
+///
+/// ```
+/// inclusion_fee = base_fee + [targeted_length_fee_adjustment * length_fee] + [targeted_weight_fee_adjustment * weight_fee];
+/// ```
+///
+/// As you can see `targeted_fee_adjustment` becomes `targeted_weight_fee_adjustment` but the behavior
+/// remains the same, the side effect is the changing to the value `targeted_length_fee_adjustment`,
+/// this formula is achievable because inside pallet_transaction_payment the function `compute_fee_raw`
+/// that just computes the final fee associated with an extrinsic uses the associated type `LenghtToFee`
+/// that converts the length of an extrinsic to a fee.
+///
+/// By default the implementation is a constant multiplication but we want to achieve a dynamic formula
+/// that can adapt based on the usage of the network, this can't solely be done by this struct but needs
+/// to be bundled with a custom implementation of `LenghtToFee`.
+///
+/// This struct ONLY provide a dynamic update of `targeted_length_fee_adjustment` and `targeted_weight_fee_adjustment`
+/// based on the congestion and usage of the blocks, while the formula si effectively implemented like
+/// explained above only thanks to `LenghtToFee`
+pub struct BlobsFeeAdjustment<T: frame_system::Config>(core::marker::PhantomData<T>);
+
+impl<T: frame_system::Config> Convert<Multiplier, Multiplier> for BlobsFeeAdjustment<T>
+where
+    T: frame_system::Config,
+{
+    /// This function should be a pure function used to update NextFeeMultiplier
+    /// but will also has the side effect of update NextLenghtMultiplier
+    fn convert(previous_fee_multiplier: Multiplier) -> Multiplier {
+        // Update TargetBlockSize if needed
+        let all_extrinsic_len = <frame_system::Pallet<T>>::all_extrinsics_len();
+
+        if all_extrinsic_len > TargetBlockSize::get() {
+            // Increase the tracker if needed
+            let tracker = BlockSizeTracker::get();
+
+            // If the used_block_size is larger than the TargetBlockSize
+            // for more than IncreaseDeltaBlocks consecutive, then the TargetBlockSize
+            // will be increased by DeltaTargetBlockSize.
```

Comment:
Why is this done?

The `TargetBlockSize` should be pretty much static, except when updated by governance.

Reply:
You can think of the fee mechanism as a "thermostat" which adjusts the temperature to encourage or discourage a certain level of use. Adjusting the target level of use is like adding the option to "open a window". A controller which can do both automatically is going to be extremely difficult to calibrate.

Reply:
I agree with you, having this variable factor would make the system more complex and difficult to calibrate, and the reason why I've done it is probably a misunderstanding of what you wrote in #16:

> If blocks are consistently larger than 0.4MB, the prices should adapt automatically. And in that case, we should also adapt, since there is very stable demand.

The thought here is something like what I did, which is to implement an automatic process to increase the economic block fullness limit if there is stable and consistent demand. That's why I defined IncreaseDeltaBlocks (the number of blocks the TargetBlockSize must be exceeded) and DeltaTargetBlockSize (the value of the increment).

From your message, I think what I've understood is not correct, and what you envisioned is just something like: if we see constant demand (as sudo) or if the governance wants it, then the `TargetBlockSize` could be increased!

I will remove this logic as soon as you confirm this! 😅

Reply:
> From your message, I think what I've understood is not correct, and what you envisioned is just something like: if we see constant demand (as sudo) or if the governance wants it, then the TargetBlockSize could be increased!

Exactly this. I see where you got the idea, though, and your thought process of creating an automated adapter. But yes, let's expect it to be manual through governance.